src/Ilios/CoreBundle/Entity/Manager/PermissionManager.php

```diff
@@ -125,6 +125,55 @@ public function userHasReadPermissionToSchool(UserInterface $user, SchoolInterfa
     /**
      * {@inheritdoc}
      */
+    public function userHasReadPermissionToSchools(UserInterface $user, ArrayCollection $schools)
+    {
+        $criteria = [
+            'tableName'     => 'schools',
+            self::CAN_READ  => true,
+            'user'          => $user,
+        ];
+
+        $schoolsWithReadPermission = $this->findPermissionBy($criteria);
+        
+        if ($schoolsWithReadPermission->count() === 0) {
+            return false;
+        }
+        
+        $overlap = array_intersect($schools->toArray(), $schoolsWithReadPermission->toArray());
+
+        return ! empty($overlap);
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function userHasWritePermissionToSchools(UserInterface $user, ArrayCollection $schools)
+    {
+        $criteria = [
+            'tableName'     => 'schools',
+            self::CAN_WRITE => true,
+            'user'          => $user,
+        ];
+
+        $schoolsWithWritePermission = $this->findPermissionBy($criteria);
+        
+        if ($schoolsWithWritePermission->count() === 0) {
+            return false;
+        }
+        
+        foreach ($schools as $school) {
```

Comment:
since you changed it to `array_intersect` above, please change it here is as well. thanks!